packages/amplify_datastore/ios/Classes/types/model/FlutterAuthRule.swift

```diff
@@ -0,0 +1,96 @@
+/*
+* Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+
+import Foundation
+import Amplify
+
+public struct FlutterAuthRule {
+    
+    private let ownerField : String?
+    private let identityClaim : String?
+    private let groupClaim : String?
+    private let groups : [String]?
+    private let groupsField : String?
+    private var authStrategy : AuthStrategy!
```

Comment:
It's weird to mix & match Flutter models and external models this way. If we are creating a layer of model in between (Native and Method Channel), it should be 100% flutter model (preferred way) or not create the middle layer model at all (Not so preferred albeit faster way, see `QueryPredicateBuilder`)

Reply:
Alright I can just store those as string then and then convert them when the convert to native method is called.

Reply:
code change added and tested